server/queries/BattleQueries.cpp

```diff
@@ -104,6 +98,12 @@ void CBattleDialogQuery::onRemoval(PlayerColor color)
 	}
 	else
 	{
+		auto hero = bi->getSideHero(BattleSide::ATTACKER);
+		auto visitingObj = bi->getDefendedTown() ? bi->getDefendedTown() : gh->getVisitingObject(hero);
```

Comment:
Why do we need actually "visiting object" instead of defender army? `bi->getSideArmy` I think?